Public/Src/Cache/ContentStore/App/Hello.cs

```diff
@@ -23,18 +23,25 @@ internal sealed partial class Application
             [Description("GRPC port on the target machine"), DefaultValue(ServiceConfiguration.GrpcDisabledPort)] int grpcPort,
             [Description("Name of the memory mapped file used to share GRPC port. 'CASaaS GRPC port' if not specified.")] string grpcPortFileName)
         {
-            Initialize();
-
-            if (grpcPort == 0)
+            try
             {
-                grpcPort = Helpers.GetGrpcPortFromFile(_logger, grpcPortFileName);
-            }
+                Initialize();
+
+                if (grpcPort == 0)
+                {
+                    grpcPort = Helpers.GetGrpcPortFromFile(_logger, grpcPortFileName);
+                }
 
-            var _channel = new Channel(host, grpcPort, ChannelCredentials.Insecure);
-            var _client = new ContentServer.ContentServerClient(_channel);
-            var helloResponse = _client.Hello(new HelloRequest(), new CallOptions(deadline: DateTime.UtcNow + TimeSpan.FromSeconds(2)));
+                var _channel = new Channel(host, grpcPort, ChannelCredentials.Insecure);
+                var _client = new ContentServer.ContentServerClient(_channel);
+                var helloResponse = _client.Hello(new HelloRequest(), new CallOptions(deadline: DateTime.UtcNow + TimeSpan.FromSeconds(2)));
 
-            _logger.Always("Hello response {0}: {1}", helloResponse.Success ? "succeeded" : "failed", helloResponse.ToString());
+                _logger.Always("Hello response {0}: {1}", helloResponse.Success ? "succeeded" : "failed", helloResponse.ToString());
+            }
+            catch (Exception e)
+            {
+                _logger.Always(e.ToString());

```

Comment:
This is good, but (1) was there some specific thing that happened that this was added to address and (2) do you need to add additional text to identify site of failure in log, so does the logger add enough for you to see where it occurred already?

Reply:
1) A failed connection will throw an exception, though I don't remember which kind of exception. That exception flowed through and was not logged.

2) This verb is not intended to ever be used in a production environment, outside testing that one can connect to an existing deployment. By default, this logs the entirety of the exception, rather than just the exception's message.

---
In reply to: [279877617](https://github.com/Microsoft/BuildXL/pull/220#discussion_r279877617) [](ancestors = 279877617)